lta/rest_server.py

```diff
@@ -87,6 +87,26 @@ def initialize(self, db: Any, check_claims: CheckClaims, *args: Any, **kwargs: A
         self.db = db
         self.check_claims = check_claims
 
+class FilesActionsBulkCreateHandler(BaseLTAHandler):
+    @lta_auth(roles=['system'])
+    async def post(self) -> None:
+        req = json_decode(self.request.body)
+        if 'files' not in req:
+            raise tornado.web.HTTPError(400, reason="missing files field")
+        if not isinstance(req['files'], list):
+            raise tornado.web.HTTPError(400, reason="files field is not a list")
+        if not req['files']:
+            raise tornado.web.HTTPError(400, reason="files field is empty")
+
+        for xfer_file in req["files"]:
+            xfer_file["uuid"] = uuid1().hex
+            xfer_file["create_timestamp"] = datetime.utcnow().isoformat()
```

Comment:
Use the `now()` function for timestamps, so we stay unified

Reply:
I think we've been using `utcnow()` instead of `now()`:

(env) pmeade@bluebird ~/github/lta $ fgrep -R "datetime.now" lta/*

(env) pmeade@bluebird ~/github/lta $ fgrep -R "datetime.now" tests/*

(env) pmeade@bluebird ~/github/lta $ fgrep -R "datetime.utcnow" lta/*
lta/log_format.py:            'timestamp': datetime.utcnow().isoformat(),
lta/picker.py:        timestamp = datetime.utcnow().isoformat()
lta/picker.py:        self.last_work_begin_timestamp = datetime.utcnow().isoformat()
lta/picker.py:        self.last_work_end_timestamp = datetime.utcnow().isoformat()
lta/picker.py:                "timestamp": datetime.utcnow().isoformat(),
lta/picker.py:            "timestamp": datetime.utcnow().isoformat()
lta/rest_server.py:    return datetime.utcnow().isoformat(timespec='seconds')
lta/rest_server.py:        cutoff_time = datetime.utcnow() - timedelta(hours=self.claim_age)
lta/rest_server.py:            xfer_file["create_timestamp"] = datetime.utcnow().isoformat()
lta/rest_server.py:        old_data = (datetime.utcnow() - timedelta(seconds=60*5)).isoformat()

(env) pmeade@bluebird ~/github/lta $ fgrep -R "datetime.utcnow" tests/*
tests/test_rest_server.py:    request = {'1.1': {'t': datetime.utcnow().isoformat(), 'foo': 'bar'}}
tests/test_rest_server.py:    request2 = {'1.2': {'t': datetime.utcnow().isoformat(), 'baz': 2}}
tests/test_rest_server.py:    request = {'2.1': {'t': (datetime.utcnow() - timedelta(hours=1)).isoformat(), 'foo': 'bar'}}

Reply:
Ahhh, there is a `now()` function in rest_server. Got it.

Reply:
If you wanted to be explicit, we could rename the function to utcnow.  But since I prefer all times in utc, it just seems like more typing. :smile: